packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,356 @@
+    bool public mRevoked;
+
+    // the time at which the revocation has taken place
+    uint256 public mRevokeTime;
+
+    /**
+     * @notice A constructor for initialising a new instance of a Vesting Schedule contract
+     * @param beneficiary address of the beneficiary to whom vested tokens are transferred
+     * @param vestingAmount the amount that is to be vested by the contract
+     * @param vestingCliff duration in seconds of the cliff in which tokens will begin to vest
+     * @param vestingStartTime the time (as Unix time) at which point vesting starts
+     * @param vestingPeriodSec duration in seconds of the period in which the tokens will vest
+     * @param vestAmountPerPeriod the vesting amound per period where period is the vestingAmount distributed over the vestingPeriodSec
+     * @param revocable whether the vesting is revocable or not
+     * @param revoker address of the person revoking the vesting
+     * @param refundDestination address of the refund receiver after the vesting is deemed revoked
+     */
+    constructor (address beneficiary,
+                uint256 vestingAmount,
+                uint256 vestingCliff,
+                uint256 vestingStartTime,
+                uint256 vestingPeriodSec,
+                uint256 vestAmountPerPeriod,
+                bool    revocable,
+                address revoker,
+                address refundDestination) public {
+
+        // do some basic checks
+        require(vestingAmount > 0, "Amount must be positive");
+        require(beneficiary != address(0), "Beneficiary is the zero address");
+        require(refundDestination != address(0), "Refund destination is the zero address");
+        require(vestingCliff <= vestingPeriodSec, "Vesting cliff is longer than duration");
+        require(vestingPeriodSec > 0, "Vesting period is 0 s.");
+        require(vestAmountPerPeriod <= vestingAmount, "Vesting amount per period is greater than the total vesting amount");
+        require(vestingStartTime.add(vestingCliff) > block.timestamp, "Final time is before current time");
+
+        //make the vesting instance an account
+        bool accountCreated = getAccounts().createAccount();
+        require(accountCreated);
+
+        mVestingPeriods = vestingAmount.div(vestAmountPerPeriod);
+        mBeneficiary = beneficiary;
+        mVestingAmount = vestingAmount;
+        mVestAmountPerPeriod = vestAmountPerPeriod;
+        mRevocable = revocable;
+        mVestingPeriodSec = vestingPeriodSec;
+        mCliffStartTime = vestingStartTime.add(vestingCliff);
+        mVestingStartTime = vestingStartTime;
+        mRefundDestination = refundDestination;
+        mRevoker = revoker;
+    }
+
+    /**
+     * @notice Transfers available released tokens from the vesting back to beneficiary.
+     */
+    function withdraw() external onlyBeneficiary {
+        uint256 releasableAmount = _getReleasableAmount(block.timestamp);
```

Comment:
nit: would replacing `release` with `withdraw` here and everywhere be more consistent?

Reply:
done